src/algorithms/highlevel/sbic.cpp

```diff
@@ -240,18 +250,40 @@ void SBic::compute()
         window = subarray(features, 0, nFeatures-1, currSeg, endSeg);
 
         // A change has been found
-        if ((i = bicChangeSearch(window, _inc1, currSeg, dmin)))
+        std::pair<int, std::vector<Real> > bicChangeResult;
+        bicChangeResult = bicChangeSearch(window, _inc1, currSeg, dmin);
+        i = bicChangeResult.first;
+        std::vector<Real> tmpBicValues(bicChangeResult.second);
+        
+        if (i)
         {
+            prevCurrSeg = currSeg;
             segmentation.push_back(i);
             segValues.push_back(dmin);
             currSeg = (i + _inc1);
             endSeg = currSeg - 1;
+            
+            int nToAdd = i - prevCurrSeg + 1;
+            
+            // Store bic values up to (and including) the peak
+            for (int j=0; j < nToAdd; ++j)
+            {
+                bicValues.push_back(tmpBicValues[j]);
```

Comment:
How large are these arrays typically? Memory could be allocated to avoid excessive push_back overhead